codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AssembleMiddlewareStack.java

```diff
@@ -0,0 +1,145 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.List;
+import software.amazon.smithy.aws.traits.auth.SigV4Trait;
+import software.amazon.smithy.aws.traits.auth.UnsignedPayloadTrait;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.ProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.traits.AuthTrait;
+import software.amazon.smithy.utils.ListUtils;
+
+public class AssembleMiddlewareStack implements GoIntegration{
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        Symbol middlewareStepAfter = SymbolUtils.createValueSymbolBuilder(
+                "After", SmithyGoDependency.SMITHY_MIDDLEWARE
+        ).build();
+
+        Symbol middlewareStepBefore = SymbolUtils.createValueSymbolBuilder(
+                "Before", SmithyGoDependency.SMITHY_MIDDLEWARE
+        ).build();
+
+        Symbol RequestInvocationIDMiddleware = SymbolUtils.createValueSymbolBuilder(
+                "RequestInvocationIDMiddleware", AwsGoDependency.AWS_MIDDLEWARE
+        ).build();
+
+        Symbol attemptClockSkewMiddleware = SymbolUtils.createValueSymbolBuilder(
+                "AttemptClockSkewMiddleware", AwsGoDependency.AWS_MIDDLEWARE
+        ).build();
+
+        Symbol newAttemptMiddleware = SymbolUtils.createValueSymbolBuilder(
+                "NewAttemptMiddleware", AwsGoDependency.AWS_RETRY_MIDDLEWARE
+        ).build();
+
+        Symbol metricHeaderMiddleware = SymbolUtils.createValueSymbolBuilder(
+                "MetricsHeaderMiddleware", AwsGoDependency.AWS_RETRY_MIDDLEWARE
+        ).build();
+
+        Symbol unsignedPayloadSignerMiddleware = SymbolUtils.createValueSymbolBuilder(
+                "UnsignedPayloadMiddleware", AwsGoDependency.AWS_V4SIGNER_MIDDLEWARE
+        ).build();
+
+        Symbol computePayloadSHA256Middleware = SymbolUtils.createValueSymbolBuilder(
+                "ComputePayloadSHA256Middleware", AwsGoDependency.AWS_V4SIGNER_MIDDLEWARE
+        ).build();
+
+        Symbol newSignHTTPRequestMiddleware = SymbolUtils.createValueSymbolBuilder(
+                "NewSignHTTPRequestMiddleware", AwsGoDependency.AWS_V4SIGNER_MIDDLEWARE
+        ).build();
+
+        return ListUtils.of(
+                // Add RequestInvocationIDMiddleware to operation stack
+                RuntimeClientPlugin.builder()
+                    .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
+                        writer.write("$L.Initialize.Add($T{}, $T)", stackOperand,
+                                RequestInvocationIDMiddleware, middlewareStepAfter);
+                    }).build(),
```

Comment:
Not saying this needs to change, but from a readability point of view, this style with everything predefined as symbols personally is much more difficult to read vs more inline code. 

It might help to go a step further with the abstraction had have helpers for the `writer.write` behavior using enums for the `Initialize`, etc statements, because from the looks of these this is the only things that are really changing between several of these statements.

The `After` could also be an enum too. or the helper already does this, e.g. 

```
writeAddMiddlewareAfter(writer, INITIALIZE_MIDDLEWARE, "&$L{}, middlewareSymbol)
````